### PR TITLE
change memory to calldata

### LINKED ISSUE #150: change memory to calldata
https://github.com/The-Poolz/LockDealNFT/blob/418b5ed16ee1fe38197a420fbc1c62fe5b354607/contracts/ProviderInterface/IProviderSingleIdRegistrar.sol#L5

### PR DESCRIPTION
**Pull request by YouStillAlive**
closes #150

### CODE REVIEW

File: contracts/Provider/BasicProvider.sol [hunk 1]
```diff
@@ -3,8 +3,9 @@ pragma solidity ^0.8.0;
 
 import "./ProviderModifiers.sol";
 import "../ProviderInterface/IProvider.sol";
+import "../ProviderInterface/IProviderSingleIdRegistrar.sol";
 
-abstract contract BasicProvider is IProvider, ProviderModifiers {
+abstract contract BasicProvider is IProvider, IProviderSingleIdRegistrar, ProviderModifiers {
```
Comment (Lomet) [inline contracts/Provider/BasicProvider.sol:8]: Why this change?

**Review by Lomet — APPROVED** [review]

File: contracts/Provider/BasicProvider.sol (hunk 1 above)
Comment (YouStillAlive) [reply contracts/Provider/BasicProvider.sol:8]: `BasicProvider` implements the `registerPool` function. `IProviderSingleIdRegistrar` declares `registerPool`
